internal/provider/resource_vault_cluster.go

```diff
@@ -56,6 +57,17 @@ func resourceVaultCluster() *schema.Resource {
 				ForceNew:         true,
 				ValidateDiagFunc: validateSlugID,
 			},
+			"tier": {
+				Description:      "Tier of the HCP Vault cluster. Valid options for tiers - `development`, `standard_small`, `standard_medium`, `standard_large`",
```

Comment:
Thanks for addressing my suggestions! ☺️ One more thing: I'd like to keep `development` as the documented option here, but right now that will result in a validation error because the API expects `dev`. I'm thinking the translation of `development` to `dev` could go in the DiffSuppressFunc - or possibly Validate if that runs first.

Reply:
I've made the translation with TODOs because in my opinion we should go with `dev` instead for now till it is changed in the API to `DEVELOPMENT`. Changing/overriding an argument in either in Validate() or DiffSuppressFunc() feels weird and having a Transform() just for this use case seems an overkill. Also checking just for a string can introduce technical debt for the future. 

I also needed a third translation  [here](https://github.com/hashicorp/terraform-provider-hcp/blob/93dcef4a2acc233006e2f2b9c1169c06992d3233/internal/provider/resource_vault_cluster.go#L322) because after creation a rerun would force a new resource as the test shows:

```
waxb@mbp:terraform-provider-hcp (vault-cluster-tier *)$ !-2
make testacc TESTARGS='-run=TestAccVaultCluster'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/... -v -run=TestAccVaultCluster -timeout 120m
?   	github.com/hashicorp/terraform-provider-hcp/internal/clients	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/consul	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/input	(cached) [no tests to run]
=== RUN   TestAccVaultClusterAdminToken
--- PASS: TestAccVaultClusterAdminToken (911.85s)
=== RUN   TestAccVaultCluster
    resource_vault_cluster_test.go:37: Step 1/4 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
        -/+ destroy and then create replacement
         <= read (data resources)
        
        Terraform will perform the following actions:
        
          # data.hcp_vault_cluster.test will be read during apply
          # (config refers to values not yet known)
         <= data "hcp_vault_cluster" "test"  {
              ~ cloud_provider             = "aws" -> (known after apply)
              ~ created_at                 = "2021-06-14T14:37:43.737Z" -> (known after apply)
              ~ hvn_id                     = "test-hvn" -> (known after apply)
              ~ id                         = "/project/11eb17a9-6649-bffd-9503-0242ac110018/hashicorp.vault.cluster/test-vault-cluster" -> (known after apply)
              + min_vault_version          = (known after apply)
              ~ namespace                  = "admin" -> (known after apply)
              ~ organization_id            = "11eb17a9-65e5-d630-9dc1-0242ac110015" -> (known after apply)
              ~ project_id                 = "11eb17a9-6649-bffd-9503-0242ac110018" -> (known after apply)
              ~ public_endpoint            = false -> (known after apply)
              ~ region                     = "us-west-2" -> (known after apply)
              ~ tier                       = "DEV" -> (known after apply)
              ~ vault_private_endpoint_url = "test-vault-cluster.private.vault.11eb17a9-6649-bffd-9503-0242ac110018.aws.hashicorp.cloud" -> (known after apply)
              + vault_public_endpoint_url  = (known after apply)
              ~ vault_version              = "v1.7.0" -> (known after apply)
                # (1 unchanged attribute hidden)
        
              + timeouts {
                  + default = (known after apply)
                }
            }
        
          # hcp_vault_cluster.test must be replaced
        -/+ resource "hcp_vault_cluster" "test" {
              ~ cloud_provider             = "aws" -> (known after apply)
              ~ created_at                 = "2021-06-14T14:37:43.737Z" -> (known after apply)
              ~ id                         = "/project/11eb17a9-6649-bffd-9503-0242ac110018/hashicorp.vault.cluster/test-vault-cluster" -> (known after apply)
              ~ namespace                  = "admin" -> (known after apply)
              ~ organization_id            = "11eb17a9-65e5-d630-9dc1-0242ac110015" -> (known after apply)
              ~ project_id                 = "11eb17a9-6649-bffd-9503-0242ac110018" -> (known after apply)
              ~ region                     = "us-west-2" -> (known after apply)
              ~ tier                       = "DEV" -> "development" # forces replacement
              ~ vault_private_endpoint_url = "test-vault-cluster.private.vault.11eb17a9-6649-bffd-9503-0242ac110018.aws.hashicorp.cloud" -> (known after apply)
              + vault_public_endpoint_url  = (known after apply)
              ~ vault_version              = "v1.7.0" -> (known after apply)
                # (3 unchanged attributes hidden)
            }
        
        Plan: 1 to add, 0 to change, 1 to destroy.
--- FAIL: TestAccVaultCluster (721.44s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-hcp/internal/provider	1633.753s
FAIL
make: *** [testacc] Error 1
```

But I understand that later changing `dev` to `development` in consumer code can cause headache. What do you think? Should we proceed with my spaghetti?

Reply:
Agreed, this does add a bit too much spaghetti. 😅 Let's go ahead and stick with what the API uses now (`dev`).

Reply:
Tests for `dev`, `standard_small` pass

```
waxb@mbp:terraform-provider-hcp (vault-cluster-tier *)$ make testacc TESTARGS='-run=TestAccVaultCluster'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/... -v -run=TestAccVaultCluster -timeout 120m
?   	github.com/hashicorp/terraform-provider-hcp/internal/clients	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/consul	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/input	(cached) [no tests to run]
=== RUN   TestAccVaultClusterAdminToken
--- PASS: TestAccVaultClusterAdminToken (710.69s)
=== RUN   TestAccVaultCluster
--- PASS: TestAccVaultCluster (757.79s)
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/provider	1468.981s
```